configurate-core/src/main/java/ninja/leaping/configurate/objectmapping/Adapter.java

```diff
@@ -16,13 +16,19 @@
  */
 package ninja.leaping.configurate.objectmapping;
 
+import ninja.leaping.configurate.objectmapping.serialize.TypeSerializer;
+
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
+/**
+ * An annotation which may be used to supply custom (de)serialization logic to specific field
+ * The field still has to be annotated via @Settings, otherwise custom field adapter will be ingored
+ */
```

Comment:
Im not a native english speaker, if you feel like this description is too brief tell me what exactly would make more sense to you.

Reply:
Something like this might be better where some of the restrictions are noted - though it's also important to note that other object mappers could operate differently:

```
Instructs the {@link ninja.leaping.configurate.objectmapping.ObjectMapper} that the serialization and deserialization value associated with the annotated field must be handled by the supplied {@link TypeSerializer} class.

<p>Consumers should note the following:</p>
<ul>
<li>The annotated field must also be annotated with the {@link Setting} attribute in order for it to be considered by the object mapper;</li>
<li>The supplied {@link TypeSerializer} class <strong>must</strong> have a public no-args constructor - violating this restriction will cause an {@link ObjectMappingException} during object mapper creation;</li>
<li>The {@link TypeSerializer} must deserialize into a type that the field can implicity accept (for example, you cannot use a {@link String} {@link TypeSerializer} on a field that is of type {@code int}) - violating this restriction will case an {@link ObjectMappingException} during serialization or deserialization.
</ul>

<p>Note that these restrictions may be different when using derived object mappers. Consult with the documentation provided by those mappers to determine if this is the case.</p>
```

I'm also not convinced `Adapter` is the right name for this. Perhaps `CustomSerializer` or `UsingSerializer` or something descriptive like that? I don't think the name `Adapter` is obvious to anyone looking for such a feature.

Reply:
I picked "Adapter" name as the origin of the idea behind this is to have similar functionality which gson provides via its Json**Adapter** . Configurate library is format independant so therefore i just stripped the "Json".

Reply:
I know that’s where it came from. My point is that just because it’s named that elsewhere doesn’t mean that it’s the right name to use.

I’m just seeking opinions.

Reply:
I like `CustomSerializer`